### PR TITLE
Replace rosbag2_format_output by direct source files linkage

### PR DESCRIPTION
**Pull request by MichaelOrlov**
- Relates https://github.com/ros2/rosbag2/pull/1480
- Replace rosbag2_format_output by direct source files linkage
- Also put `format_service_info(..)` and `format_bag_meta_data(..)` under the rosbag2_py namespace

### CODE REVIEW

File: rosbag2_py/CMakeLists.txt [hunk 1]
```diff
@@ -73,11 +59,12 @@ target_link_libraries(_reader PUBLIC
 
 pybind11_add_module(_storage SHARED
   src/rosbag2_py/_storage.cpp
+  src/rosbag2_py/format_bag_metadata.hpp
```
Comment (Barry-Xu-2018) [inline rosbag2_py/CMakeLists.txt]: I think header file should not be included.

File: rosbag2_py/CMakeLists.txt [hunk 2]
```diff
@@ -91,11 +78,14 @@ target_link_libraries(_writer PUBLIC
 
 pybind11_add_module(_info SHARED
   src/rosbag2_py/_info.cpp
+  src/rosbag2_py/format_bag_metadata.hpp
```
Comment (Barry-Xu-2018) [inline rosbag2_py/CMakeLists.txt]: Remove it.
The cause is the same as above comments.

File: rosbag2_py/CMakeLists.txt [hunk 3]
```diff
@@ -91,11 +78,14 @@ target_link_libraries(_writer PUBLIC
...
+  src/rosbag2_py/format_bag_metadata.cpp
+  src/rosbag2_py/format_service_info.hpp
```
Comment (Barry-Xu-2018) [inline rosbag2_py/CMakeLists.txt]: Remove it.

**Comment by Barry-Xu-2018** [comment]
rosbag2_py/src/rosbag2_py/visibility_control.hpp is unused now. It is added in my last commit `Use visibility control for format library`.
It should be removed.

**Comment by MichaelOrlov** [comment]
@Barry-Xu-2018 I've addressed your comments

**Pull request merged**